File: client-clojure/src/sirqul_io_t_platform/specs/leg_response.clj
(ns sirqul-io-t-platform.specs.leg-response
  (:require [clojure.spec.alpha :as s]
            [spec-tools.data-spec :as ds]
            [sirqul-io-t-platform.specs.name-string-value-response :refer :all]
            [sirqul-io-t-platform.specs.step-response :refer :all]
            )
  (:import (java.io File)))


(def leg-response-data
  {
   (ds/opt :valid) boolean?
   (ds/opt :message) string?
   (ds/opt :version) float?
   (ds/opt :serializeNulls) boolean?
   (ds/opt :startTimeLog) int?
   (ds/opt :errorCode) string?
   (ds/opt :request) (s/coll-of name-string-value-response-spec)
   (ds/opt :appKey) string?
   (ds/opt :distance) float?
   (ds/opt :duration) int?
   (ds/opt :startLatitude) float?
   (ds/opt :startLongitude) float?
   (ds/opt :startAltitude) float?
   (ds/opt :startDate) int?
   (ds/opt :endLatitude) float?
   (ds/opt :endLongitude) float?
   (ds/opt :endAltitude) float?
   (ds/opt :endDate) int?
   (ds/opt :created) int?
   (ds/opt :tags) string?
   (ds/opt :deviceId) string?
   (ds/opt :accountId) int?
   (ds/opt :deviceSignature) string?
   (ds/opt :parentAccountId) int?
   (ds/opt :steps) (s/coll-of step-response-spec)
   (ds/opt :type) string?
   (ds/opt :returning) string?
   })

(def leg-response-spec
  (ds/spec
    {:name ::leg-response
     :spec leg-response-data}))

File: client-scalaz/src/main/scala/org/openapitools/client/api/LegResponse.scala
package org.openapitools.client.api

import argonaut._
import argonaut.EncodeJson._
import argonaut.DecodeJson._

import org.http4s.{EntityDecoder, EntityEncoder}
import org.http4s.argonaut._
import org.joda.time.DateTime


import LegResponse._

case class LegResponse (
  valid: Option[Boolean],
message: Option[String],
version: Option[Double],
serializeNulls: Option[Boolean],
startTimeLog: Option[Long],
errorCode: Option[String],
request: Option[List[NameStringValueResponse]],
appKey: Option[String],
distance: Option[Double],
duration: Option[Long],
startLatitude: Option[Double],
startLongitude: Option[Double],
startAltitude: Option[Double],
startDate: Option[Long],
endLatitude: Option[Double],
endLongitude: Option[Double],
endAltitude: Option[Double],
endDate: Option[Long],
created: Option[Long],
tags: Option[String],
deviceId: Option[String],
accountId: Option[Long],
deviceSignature: Option[String],
parentAccountId: Option[Long],
steps: Option[List[StepResponse]],
`type`: Option[String],
returning: Option[String])

object LegResponse {
  import DateTimeCodecs._

  implicit val LegResponseCodecJson: CodecJson[LegResponse] = CodecJson.derive[LegResponse]
  implicit val LegResponseDecoder: EntityDecoder[LegResponse] = jsonOf[LegResponse]
  implicit val LegResponseEncoder: EntityEncoder[LegResponse] = jsonEncoderOf[LegResponse]
}
